android-agent/src/main/java/io/opentelemetry/android/config/OtelRumConfig.java

```diff
@@ -121,20 +120,6 @@ public OtelRumConfig setDiskBufferingConfiguration(
         return this;
     }
 
-    /**
-     * Sets the configuration so that network change monitoring, which is enabled by default, will
-     * not be started.
-     */
-    public OtelRumConfig disableNetworkChangeMonitoring() {
-        networkChangeMonitoringEnabled = false;
-        return this;
-    }
-
-    /** Returns true if network change monitoring is enabled (default = true). */
-    public boolean isNetworkChangeMonitoringEnabled() {
-        return networkChangeMonitoringEnabled;
-    }
-
```

Comment:
So now the configuration choice is no longer programmatic and cannot be dynamic. It is determined statically at build-time depending on whether or not the user has included the network instrumentation dependency?

I think that's fine, I just wanted to think that through, out loud. If/when the need for additional configuration arises, I suppose we can handle that within each instrumentation module that requires it.

Reply:
> So now the configuration choice is no longer programmatic and cannot be dynamic. It is determined statically at build-time depending on whether or not the user has included the network instrumentation dependency?

These changes mean that the core config is not aware of the network changes instrumentation. As it is now, it can only be enabled at build-time by adding the instrumentation dependency and it cannot be disabled if it's in the classpath at runtime. However, we could expand `AndroidInstrumentation` to allow for disabling an instrumentation, even if it was already enabled (by using the [suppress context key](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9739)) - Or we could also avoid loading instrumentations altogether from the classpath, and instead require each instrumentation to be manually "enabled" by registering it through the `AndroidInstrumentaionRegistry`. So yeah it's as you mentioned, I think we can handle different use cases if the need arises.

Reply:
Cool. I think it could be helpful to have a tracking issue for this, so I've opened #411 to track this. 👍🏻